### PR TITLE
rollback on db error

### PR DESCRIPTION
**Pull request by jshom**
**Issue**
Used to hang on incomplete/err transaction:

Prior Error Logs with Hang:
```
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:03:15] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:03:15 +0000] "GET /api/error HTTP/1.1" 200 116 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:03:16.628 UTC [49] ERROR:  current transaction is aborted, commands ignored until end of transaction block
yacs_db       | 2020-11-07 16:03:16.628 UTC [49] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: current transaction is aborted, commands ignored until end of transaction block
yacs_api      |
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:03:16] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:03:16 +0000] "GET /api/error HTTP/1.1" 200 116 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:03:17.015 UTC [49] ERROR:  current transaction is aborted, commands ignored until end of transaction block
yacs_db       | 2020-11-07 16:03:17.015 UTC [49] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: current transaction is aborted, commands ignored until end of transaction block
yacs_api      |
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:03:17] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:03:17 +0000] "GET /api/error HTTP/1.1" 200 116 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:03:17.424 UTC [49] ERROR:  current transaction is aborted, commands ignored until end of transaction block
```

New Error Logs without Hang:
```
yacs_api      | [INFO] Database Connected
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:03:01] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:03:01 +0000] "GET /api/error HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:03:01.726 UTC [44] ERROR:  relation "a_table_that_doesnt_exist" does not exist at character 15
yacs_db       | 2020-11-07 16:03:01.726 UTC [44] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: relation "a_table_that_doesnt_exist" does not exist
yacs_api      | LINE 1: select * from a_table_that_doesnt_exist;
yacs_api      |                       ^
yacs_api      |
yacs_api      | [INFO] Database Connected
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:03:01] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:03:01 +0000] "GET /api/error HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:03:02.222 UTC [45] ERROR:  relation "a_table_that_doesnt_exist" does not exist at character 15
yacs_db       | 2020-11-07 16:03:02.222 UTC [45] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: relation "a_table_that_doesnt_exist" does not exist
yacs_api      | LINE 1: select * from a_table_that_doesnt_exist;
yacs_api      |                       ^
yacs_api      |
yacs_api      | [INFO] Database Connected
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:03:02] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:03:02 +0000] "GET /api/error HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:03:02.710 UTC [46] ERROR:  relation "a_table_that_doesnt_exist" does not exist at character 15
yacs_db       | 2020-11-07 16:03:02.710 UTC [46] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: relation "a_table_that_doesnt_exist" does not exist
yacs_api      | LINE 1: select * from a_table_that_doesnt_exist;
```

To test I added sample route to /src/api/app.py under `/api/error` to force a db error
```py
  @app.route('/api/error')
  def broken_query():
      res = db_conn.execute('select * from a_table_that_doesnt_exist;', (), True)
      return Response(str(res))
```

### CODE REVIEW

**Review by RichtXO — APPROVED** [review]

File: src/api/db/connection.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ def execute(self, sql, args, isSELECT=True):
         except psycopg2.Error as e:
             print("DATABASE ERROR: ", end="")
             print(e)
+            self.conn.close()
```
Comment (123joshuawu) [inline src/api/db/connection.py:42]: i think `self.conn.rollback()` would prolly be better than closing/reconnecting, or we could just run this and open a tech debt issue

**Review by marchdan — APPROVED** [review]

File: src/api/db/connection.py (hunk 1 above)
Comment (jshom) [reply src/api/db/connection.py:42]: will give it a testeroo, agree

Comment (jshom) [reply src/api/db/connection.py]: works and logs look happy

**Review by 123joshuawu — APPROVED** [review]
hopefully this works lol

**Comment by jshom** [comment]
no hoping logs look happy:
```
aracter 15
yacs_db       | 2020-11-07 16:26:05.267 UTC [28] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: relation "a_table_that_doesnt_exist" does not exist
yacs_api      | LINE 1: select * from a_table_that_doesnt_exist;
yacs_api      |                       ^
yacs_api      |
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:26:05] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:26:05 +0000] "GET /api/error HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:26:05.829 UTC [28] ERROR:  relation "a_table_that_doesnt_exist" does not exist at character 15
yacs_db       | 2020-11-07 16:26:05.829 UTC [28] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: relation "a_table_that_doesnt_exist" does not exist
yacs_api      | LINE 1: select * from a_table_that_doesnt_exist;
yacs_api      |                       ^
yacs_api      |
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:26:05] "GET /api/error HTTP/1.0" 200 -
yacs_nginx    | 172.20.0.1 - - [07/Nov/2020:16:26:05 +0000] "GET /api/error HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0" "-"
yacs_db       | 2020-11-07 16:26:06.248 UTC [28] ERROR:  relation "a_table_that_doesnt_exist" does not exist at character 15
yacs_db       | 2020-11-07 16:26:06.248 UTC [28] STATEMENT:  select * from a_table_that_doesnt_exist;
yacs_api      | DATABASE ERROR: relation "a_table_that_doesnt_exist" does not exist
yacs_api      | LINE 1: select * from a_table_that_doesnt_exist;
yacs_api      |                       ^
yacs_api      |
yacs_api      | 172.20.0.5 - - [07/Nov/2020 16:26:06] "GET /api/error HTTP/1.0" 200 -
```

**Pull request merged**